### PR TITLE
No CURL, McCabe<11, new job descriptions, logs made clean

### PR DESCRIPTION
**Pull request by complynx**
Finally, I managed to rewrite this. It's not yet in it's perfect state, but it's already some.
**TODO** — test it. I have right now no environment to make that.
It might fail, but the whole thing now is rather simple and wont take time to fix.

### CODE REVIEW

**Comment by mlassnig** [comment]
Thanks a lot! I'll have a look immediately.

By the way, next time please communicate before doing a patch :-) Otherwise I might have changed things in the meantime, and I end up with merging errors.

**Comment by complynx** [comment]
I'm thinking of pull-requests to be that kind of message, and straight commits as the message for «I'm onto something»

**Comment by mlassnig** [comment]
Hmmmm, I only see the pull request once you actually have commited something. You might work on a feature without Paul/me knowing before that, unless you make a placeholder pullrequest/commit. Just a simple email stating "Hey, gonna work on this feature for the next few days, stay tuned" is enough. Thanks.

**Comment by complynx** [comment]
Ok, the next will be like that. Sorry for the inconvenience.

**Comment by mlassnig** [comment]
No problem, and thanks again for the code!

File: pilot.py [hunk 1]
```diff
@@ -65,6 +66,18 @@ def main():
                             required=True,
                             help='MANDATORY: queue name (e.g., AGLT2_TEST-condor')
 
+    # SSL certificates
+    arg_parser.add_argument('--cacert',
+                            dest='cacert',
+                            default=None,
```
Comment (mlassnig) [inline pilot.py:73]: the default here should be the contents of $X509_USER_PROXY

File: pilot.py [hunk 2]
```diff
@@ -65,6 +66,18 @@ def main():
...
+                            help='CA certificate to use with HTTPS calls to server, commonly X509 proxy',
+                            metavar="path/to/your/certificate")
+    arg_parser.add_argument('--capath',
+                            dest='capath',
+                            default=None,
```
Comment (mlassnig) [inline pilot.py:78]: the default here should be /etc/grid-security/certificates

File: pilot.py (hunk 1 above)
Comment (complynx) [reply pilot.py:73]: See the util.https, there is the defaults finder. It is slightly greater.

File: pilot.py (hunk 2 above)
Comment (complynx) [reply pilot.py:78]: Saem

File: pilot.py (hunk 1 above)
Comment (complynx) [reply pilot.py:73]: I was doing at-the-init defaults-obtaining thing, but it only works in 2.7 and higher, with use of some lambdas, see pilot-2.0 code.

**Comment by mlassnig** [comment]
you forgot to add yourself as author to all the files you touched ;-)

File: pilot.py (hunk 1 above)
Comment (mlassnig) [reply pilot.py:73]: ok

**Comment by complynx** [comment]
I always forget that =)

**Pull request merged**